packages/features/ee/impersonation/lib/ImpersonationProvider.ts

```diff
@@ -85,6 +86,26 @@ const ImpersonationProvider = CredentialsProvider({
     checkUserIdentifier(creds);
     checkPermission(session);
 
+    let TeamWhereClause: Prisma.MembershipWhereInput = {
+      disableImpersonation: false, // Ensure they have impersonation enabled
+      accepted: true, // Ensure they are apart of the team and not just invited.
+      team: {
+        id: teamId, // Bring back only the right team
+      },
+    };
+
+    // If you are an admin we dont need to follow this flow -> We can just follow the usual flow
+    // If orgId and teamId are the same we can follow the same flow
+    if (session?.user.org?.id && session.user.org.id !== teamId && session?.user.role !== "ADMIN") {
+      TeamWhereClause = {
+        disableImpersonation: false,
+        accepted: true,
+        team: {
+          id: session.user.org.id,
+        },
+      };
+    }
+
     // Get user who is being impersonated
     const impersonatedUser = await prisma.user.findFirst({
```

Comment:
what do you think about putting this logic in a function to scope it out of the main handler ? I see something like this 

```
const getImpersonatedUser = async ({
  user,
  creds,
  teamId,
}: {
  user?: Session["user"];
  teamId?: number;
  creds: Record<"teamId" | "username", string> | undefined;
}) => { ... }

....

const impersonatedUser = getImpersonatedUser(...)
 ```